unit-tests/unit-tests-common.cpp

```diff
@@ -594,9 +595,9 @@ rs2::depth_sensor restart_first_device_and_return_depth_sensor( const rs2::conte
     REQUIRE_NOTHROW( serial = dev.get_info( RS2_CAMERA_INFO_SERIAL_NUMBER ) );
     // forcing hardware reset to simulate device disconnection
     auto shared_dev = std::make_shared< rs2::device >( devices_list.front() );
-    shared_dev
-        = do_with_waiting_for_camera_connection( ctx, shared_dev, serial, [&]() { shared_dev->hardware_reset(); } );
-    rs2::depth_sensor depth_sensor = dev.query_sensors().front();
+    shared_dev = do_with_waiting_for_camera_connection( ctx, shared_dev, serial, [&]() { shared_dev->hardware_reset(); } );
+    REQUIRE( shared_dev );
+    rs2::depth_sensor depth_sensor = shared_dev->query_sensors().front();
```

Comment:
What is this? A separate bug fix?

Reply:
Yes, this fixes failures I had with the HDR test. We also so it sometimes in the CI but the options-watcher querying all options on initialization changed the timing and made it repeatable.

The bug was that after restart they continued to use the old device instead of the new one. `hdr_config` object wrongly showed that HDR is enabled after restart.

Reply:
OK, please update the description (and preferably open a ticket so this is tracked) to include this fix.

Reply:
Updated the description